nixos/modules/services/networking/knot.nix

```diff
@@ -142,7 +141,35 @@ let
 in {
   options = {
     services.knot = {
-      enable = mkEnableOption (lib.mdDoc "Knot authoritative-only DNS server");
+      enable = mkEnableOption "Knot authoritative-only DNS server";
+
+      enableXDP = mkOption {
```

Comment:
Shouldn't this be readOnly since setting it to true manually does nothing?

Reply:
As long as we allow `settingsFIle` this option can be used to relax the system hardening as required for XDP to be set up.